packages/mdx-embed/src/components/twitch/twitch.test.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { render, screen, act } from '@testing-library/react';
+
+import { Twitch } from './';
+
+const PARENT_PROP = 'http://localhost:6006/';
+
+describe('<Twitch />', () => {
+  const intersectionObserverMock = () => ({
```

Comment:
Hey @ekafyi  thanks for this. The `intersectionObserverMock` used in General Observer is a special case. There's a more common method that we're using for the component unit tests called `triggerGeneralObserver` check the [CodePen](https://github.com/PaulieScanlon/mdx-embed/blob/main/packages/mdx-embed/src/components/codepen/codepen.test.tsx) unit test for usage

Reply:
Hi @PaulieScanlon,

Without lines 9 through 16 (`intersectionObserverMock` and the `beforeEach` function), only the first test passes; the remaining two tests fail with this error:

>  Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.

I added `intersectionObserverMock`, which seemed to fix that issue.

Would you prefer removing the second and third tests (so we only have one test, _'it renders the component when provided a twitchId'_); or having all three tests with `intersectionObserverMock`; or any other solution?

Reply:
@ekafyi thanks for this! Hmmm. that's interesting. Perhaps there's an issue with our mock observer. Leave it with me and i'll have a quick look and see if i can find the problem.